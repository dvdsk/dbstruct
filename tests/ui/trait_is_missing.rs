use dbstruct::dbstruct;

struct CustomType {
    field: u8,
}

#[dbstruct]
struct Test {
    #[dbstruct(Default)]
    field: CustomType,
}

fn main() {}
