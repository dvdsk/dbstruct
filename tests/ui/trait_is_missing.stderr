error[E0277]: the trait bound `CustomType: serde::ser::Serialize` is not satisfied
  --> tests/ui/trait_is_missing.rs:10:12
   |
10 |     field: CustomType,
   |            ^^^^^^^^^^ the trait `serde::ser::Serialize` is not implemented for `CustomType`
   |
   = help: the following other types implement trait `serde::ser::Serialize`:
             &'a T
             &'a mut T
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
             (T0, T1, T2, T3, T4, T5)
           and 126 others
note: required by a bound in `DefaultTrait`
  --> src/wrappers/default_trait.rs
   |
   |     T: Serialize + DeserializeOwned + Default,
   |        ^^^^^^^^^ required by this bound in `DefaultTrait`

error[E0277]: the trait bound `CustomType: Default` is not satisfied
  --> tests/ui/trait_is_missing.rs:10:12
   |
10 |     field: CustomType,
   |            ^^^^^^^^^^ the trait `Default` is not implemented for `CustomType`
   |
note: required by a bound in `DefaultTrait`
  --> src/wrappers/default_trait.rs
   |
   |     T: Serialize + DeserializeOwned + Default,
   |                                       ^^^^^^^ required by this bound in `DefaultTrait`
help: consider annotating `CustomType` with `#[derive(Default)]`
   |
3  | #[derive(Default)]
   |

error[E0277]: the trait bound `for<'de> CustomType: serde::de::Deserialize<'de>` is not satisfied
  --> tests/ui/trait_is_missing.rs:10:12
   |
10 |     field: CustomType,
   |            ^^^^^^^^^^ the trait `for<'de> serde::de::Deserialize<'de>` is not implemented for `CustomType`
   |
   = help: the following other types implement trait `serde::de::Deserialize<'de>`:
             &'a Path
             &'a [u8]
             &'a str
             ()
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
             (T0, T1, T2, T3, T4)
           and 134 others
   = note: required because of the requirements on the impl of `serde::de::DeserializeOwned` for `CustomType`
note: required by a bound in `DefaultTrait`
  --> src/wrappers/default_trait.rs
   |
   |     T: Serialize + DeserializeOwned + Default,
   |                    ^^^^^^^^^^^^^^^^ required by this bound in `DefaultTrait`
